### PR TITLE
Update some test form in order to work properly on qt

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
There's some restriction on QT/QML for running javascript. (https://doc.qt.io/archives/qt-5.6/qtqml-javascript-hostenvironment.html) Some variable definitions in `setUP`  doesn't work/ and keyword `var` shouldn't be missed in QT. I've tested that the global variable doesn't work. Instead, I've added function to set variables, then make every test function call it.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)
JavaScript Host Environment (QtQML)
https://doc.qt.io/archives/qt-5.6/qtqml-javascript-hostenvironment.html

### CODE REVIEW

File: js/test/durfmt/testdurfmt2.js [hunk 1]
```diff
@@ -731,14 +852,24 @@ module.exports.testdurfmt2 = {
     testDurFmt_el_GR: function(test) {
         test.expect(16);
         // 1,17
+        var length = [], textfmt;
+
+        length[0] = "full";
+        length[1] = "long";
+        length[2] = "medium";
+        length[3] = "short";
```
Comment (ehoogerbeets) [inline js/test/durfmt/testdurfmt2.js]: Make "length" a global variable

File: js/test/daterange/testdatefmtrange_fmttemplate.js [hunk 2]
```diff
@@ -28,36 +28,38 @@ module.exports.testdaterangefmt = {
...
-            templatec30[i] = fmt[i].dateFmt.formats.range["c30"][abbrLength[i]];
+        var length = [], abbrLength = [], fmt;
+
+        length[0] = "full";
+        length[1] = "long";
+        length[2] = "medium";
+        length[3] = "short";
+
+        abbrLength[0] = "f";
+        abbrLength[1] = "l";
```
Comment (ehoogerbeets) [inline js/test/daterange/testdatefmtrange_fmttemplate.js]: Make abbrLength a global variable as well

Comment (gouniLee) [reply js/test/daterange/testdatefmtrange_fmttemplate.js]: There's some restriction on QT/QML for running javascript. (https://doc.qt.io/archives/qt-5.6/qtqml-javascript-hostenvironment.html) I've tested that the global variable doesn't work. Instead, I've added  `function` to set variables, then make every test function call it.

**Comment by ehoogerbeets** [comment]
All the tests pass on Qt in the development branch at the moment... Why are these changes necessary?

**Comment by gouniLee** [comment]
While I'm checking files, I noticed that many tests haven't run on QT..  If you see the line 
 https://github.com/iLib-js/iLib/blob/development/qt/NodeunitTest/TestEnvironment.qml#L29 
Tests have not been run in the meantime.
For example, date features, it runs 9,365 assertions for now.  but actually, it should have 341,464 assertions. If I fix them correctly, https://github.com/iLib-js/iLib/pull/181/commits/5c791173cf32de36fcc300d2ce6f2a35dbae9578 There are more failures.. on maps, root and phone features. There have been problems that have not been discovered.

**Comment by gouniLee** [comment]
There are additional fixes. I'd like to discuss my changes.
* Fix in `root/testrequries.js`: `require` statement qt in iLib is implemented by itself. In order to pass require test case for QT, iLib need to implement additional cases on `ilib-qt.js.` Instead of this, I just disable those case for QT. 
* Fix in `root/testglobal.js`:  The way of `require` implementation for qt doesen't fit for loading 'package.json. only for module.export form.
* Fix in `maps/testcharmapasync.js`: The `Uint8Array` which used in `mapToNative()` doesn't support on QT 5.9. It's new in ES2017. It will be enabled when QT support this feature.

With these changes, finally, all of test case is passed on QT 5.9 . Please see the difference in the number of assertions compares the `development` branch.

```
test.all:
     [exec] qml: <<<<< Start time of full test: 15:41:33 >>>>>
     [exec] qml: [runner.js] runTests: running suite /address/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /address/testSuiteFiles.js (moduleName): address
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 112065 tests were performed.
     [exec] qml: [Result] 696925 assertions of 696925 passed, 0 failed
     [exec] qml: [Result] Test completed in 2.335 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /calendar/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /calendar/testSuiteFiles.js (moduleName): calendar
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 13556 tests were performed.
     [exec] qml: [Result] 75296 assertions of 75296 passed, 0 failed
     [exec] qml: [Result] Test completed in 7.077 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /collate/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /collate/testSuiteFiles.js (moduleName): collate
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 13706 tests were performed.
     [exec] qml: [Result] 178705 assertions of 178705 passed, 0 failed
     [exec] qml: [Result] Test completed in 2009.169 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /ctype/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /ctype/testSuiteFiles.js (moduleName): ctype
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 544 tests were performed.
     [exec] qml: [Result] 1119 assertions of 1119 passed, 0 failed
     [exec] qml: [Result] Test completed in 0.028 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /date/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /date/testSuiteFiles.js (moduleName): date
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 163967 tests were performed.
     [exec] qml: [Result] 341464 assertions of 341464 passed, 0 failed
     [exec] qml: [Result] Test completed in 8.946 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /daterange/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /daterange/testSuiteFiles.js (moduleName): daterange
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 34074 tests were performed.
     [exec] qml: [Result] 72043 assertions of 72043 passed, 0 failed
     [exec] qml: [Result] Test completed in 21.335 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /durfmt/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /durfmt/testSuiteFiles.js (moduleName): durfmt
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 3194 tests were performed.
     [exec] qml: [Result] 39522 assertions of 39522 passed, 0 failed
     [exec] qml: [Result] Test completed in 17.94 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /maps/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /maps/testSuiteFiles.js (moduleName): maps
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 328 tests were performed.
     [exec] qml: [Result] 4799 assertions of 4799 passed, 0 failed
     [exec] qml: [Result] Test completed in 0.007 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /name/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /name/testSuiteFiles.js (moduleName): name
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 32940 tests were performed.
     [exec] qml: [Result] 62710 assertions of 62710 passed, 0 failed
     [exec] qml: [Result] Test completed in 1.857 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /number/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /number/testSuiteFiles.js (moduleName): number
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 4216 tests were performed.
     [exec] qml: [Result] 13936 assertions of 13936 passed, 0 failed
     [exec] qml: [Result] Test completed in 0.12 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /phone/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /phone/testSuiteFiles.js (moduleName): phone
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 36721 tests were performed.
     [exec] qml: [Result] 69646 assertions of 69646 passed, 0 failed
     [exec] qml: [Result] Test completed in 2.025 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /root/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /root/testSuiteFiles.js (moduleName): root
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 4542 tests were performed.
     [exec] qml: [Result] 9749 assertions of 9749 passed, 0 failed
     [exec] qml: [Result] Test completed in 0.041 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /strings-ext/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /strings-ext/testSuiteFiles.js (moduleName): strings-ext
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 37097 tests were performed.
     [exec] qml: [Result] 222691 assertions of 222691 passed, 0 failed
     [exec] qml: [Result] Test completed in 5.811 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /units/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /units/testSuiteFiles.js (moduleName): units
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 1955 tests were performed.
     [exec] qml: [Result] 3242 assertions of 3242 passed, 0 failed
     [exec] qml: [Result] Test completed in 4.43 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: [runner.js] runTests: running suite /util/testSuiteFiles.js
     [exec] qml: [TestEnvironment.qml] (path) : /util/testSuiteFiles.js (moduleName): util
     [exec] qml: ===========================================================================================
     [exec] qml: [Result] A total of 461 tests were performed.
     [exec] qml: [Result] 541 assertions of 541 passed, 0 failed
     [exec] qml: [Result] Test completed in 0.013 seconds
     [exec] qml: ===========================================================================================
     [exec] qml: <<<<< End time of full test: 16:57:34 >>>>>
     [exec] qml: <<<<< Total Elapse Time: 4561.283 seconds  >>>>>
     [exec] qml: 
     [exec]  *************************** All iLib tests on QML are done. ***********************************

```

**Pull request merged**